droidmediabuffer.cpp

```diff
@@ -114,10 +114,31 @@ DroidMediaBuffer *droid_media_buffer_create_from_raw_data(uint32_t w, uint32_t h
 							  DroidMediaBufferCallbacks *cb)
 {
   void *addr = NULL;
-  uint32_t width, height, dstStride;
+  uint32_t linesY, linesUV,
+           lineWidthY, lineWidthUV,
+           dstStrideY, dstStrideUV;
   uint8_t *dst;
   uint8_t *src;
 
+  // Make sure the format is supported before continue
+  switch (format) {
+  case HAL_PIXEL_FORMAT_RGBA_8888:
+  case HAL_PIXEL_FORMAT_RGBX_8888:
+  case HAL_PIXEL_FORMAT_BGRA_8888:
+  case HAL_PIXEL_FORMAT_RGB_888:
+  case HAL_PIXEL_FORMAT_RGB_565:
+  case HAL_PIXEL_FORMAT_YCbCr_422_I:
+  case HAL_PIXEL_FORMAT_YV12:
+  case HAL_PIXEL_FORMAT_YCbCr_422_SP:
+  case HAL_PIXEL_FORMAT_YCrCb_420_SP:
+  case /* QOMX_COLOR_FormatYUV420PackedSemiPlanar32m */ 0x7FA30C04:
```

Comment:
These two are set in droidmediaconstants already. Would be better to use that rather than define them twice.

Reply:
Unfortunately, they're passed to gst-droid side without a way to retrieve it from the other part of droidmedia.

https://github.com/sailfishos/droidmedia/blob/52143f26ff6dc2346cf4f14792c175493cf529b1/droidmediaconstants.cpp#L80

The `c` variable here is passed from gst-droid and is stored nowhere within droidmedia.

Reply:
Well, there must be some way to define these in one place then use them everywhere. They occur three times each in this file.

Reply:
Hmm, what's the earliest version of Android droidmedia has to support? These 2 variables seem to be defined in AOSP's OMX_IVCommon.h since KitKat.

https://android.googlesource.com/platform/frameworks/native/+/kitkat-release/include/media/openmax/OMX_IVCommon.h#162

Reply:
And, in fact, will we ever have to create those vendor-specific color formats?

Reply:
Yeah, they were added because some Qualcomm devices use them for camera buffers iirc. Droidmedia still supports the Jolla1 with its 4.1 Android base, but MTK devices are also missing these defines.